### PR TITLE
Milestone 8

### PR DESCRIPTION
**Pull request by Trictonicmp**
#### In this milestone we:
- [x] Make sure that the N+1 problem is solved when fetching all posts and their comments for a user. 
- [x]  Use Capybara to write integration tests for each view in your project.
  
- [x] User index page:
  - [x] I can see the username of all other users.
  - [x] I can see the profile picture for each user.
  - [x] I can see the number of posts each user has written.
  - [x] When I click on a user, I am redirected to that user's show page.
  
- [x] user show page:
  - [x] I can see the user's profile picture.
  - [x] I can see the user's username.
  - [x] I can see the number of posts the user has written.
  - [x] I can see the user's bio.
  - [x] I can see the user's first 3 posts.
  - [x] I can see a button that lets me view all of a user's posts.
  - [x] When I click a user's post, it redirects me to that post's show page.
  - [x] When I click to see all posts, it redirects me to the user's post's index page.

- [x] User post index page:
  - [x] I can see the user's profile picture.
  - [x] I can see the user's username.
  - [x] I can see the number of posts the user has written.
  - [x] I can see a post's title.
  - [x] I can see some of the post's body.
  - [x] I can see the first comments on a post.
  - [x] I can see how many comments a post has.
  - [x] I can see how many likes a post has.
  - [x] I can see a section for pagination if there are more posts than fit on the view.
  - [x] When I click on a post, it redirects me to that post's show page.

- [x] Post show page:
  - [x] I can see the post's title.
  - [x] I can see who wrote the post.
  - [x] I can see how many comments it has.
  - [x] I can see how many likes it has.
  - [x] I can see the post body.
  - [x] I can see the username of each commentor.
  - [x] I can see the comment each commentor left.

### CODE REVIEW

File: spec/features/home_spec.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    visit('http://localhost:3000/')
+    expect(page).to have_content('Number of posts: 4')
+    expect(page).to have_content('Number of posts: 0')
+    expect(page).to have_content('Number of posts: 5')
+  end
+
+  it 'redirects to the user/show page' do
+    visit('http://localhost:3000/')
+    expect(page).to have_link('Lilly', href: '/users/1')
+  end
```
Comment (AnselemOdims) [inline spec/features/home_spec.rb]: - [x] Please kindly simulate clicking a link to assert that it redirects to the user show page.

File: spec/features/users_spec.rb [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  it 'redirects to the user/post/show page when clicking on a post' do
+    visit('http://localhost:3000/users/1')
+    expect(page).to have_link('My forth post', href: '/users/1/posts/9')
+  end
+
+  it 'redirects to users/posts index when clicking in "See all posts"' do
+    visit('http://localhost:3000/users/1')
+    expect(page).to have_link('', href: '/users/1/posts')
+  end
+end
```
Comment (AnselemOdims) [inline spec/features/users_spec.rb]: - [x] Please kindly include tests to assert that it actually 
  - redirects to the post's show page when you click on a user's post.
  - redirects to the user's post's index page when you click on a **see all posts**.

File: spec/features/posts_spec.rb [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+  it 'has a pagination button' do
+    visit('http://localhost:3000/users/1/posts')
+    expect(page).to have_selector(:link_or_button, 'pagination')
+  end
+
+  it 'redirects to the user/post/show page when clicking on a post' do
+    visit('http://localhost:3000/users/1/posts')
+    expect(page).to have_link('My forth post', href: '/users/1/posts/9')
+  end
```
Comment (AnselemOdims) [inline spec/features/posts_spec.rb]: - [x] As stated above, please kindly include proper tests that simulate actually clicking a link

**Review by AnselemOdims — CHANGES_REQUESTED** [review]
### Hi @Trictonicmp , 
You have been doing a tremendous job so far :ok_hand:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
No linter errors :heavy_check_mark:

## Required Changes ♻️
- [ ] Please also check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by Trictonicmp** [comment]
Hi @AnselemOdims I already did the changes but for some reason seems that the status of the review is still as pending, can you check if you can submit it as changes required? to be able to submit it again.
Thank you in advance!

**Review by smunozmo — APPROVED** [review]
Hi @Trictonicmp, 👋 

## Highlights 👏 

- All integration tests are done correctly, congrats on it! 🎉 
- Excellent job using Capybara! 😎 
- Great work using the right GitHub Flow.
- Your README looks professional, well done!
- Commits are descriptive, good work.
- Linters are working properly, great job!

Your project is complete!  There is nothing else to say other than... it's time to merge it! 🚀 
Very nice work, congratulations! 🎉 

# APPROVED! 💯  💯  💯 




Cheers and Happy coding! 🤓 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

✅  Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Trictonicmp** [comment]
Thank you so much @AnselemOdims!!!

**Pull request merged**